src/main/scala/mesosphere/marathon/core/instance/Goal.scala

```diff
@@ -8,10 +8,21 @@ import play.api.libs.json.{Format, JsError, JsResult, JsString, JsSuccess, JsVal
   * Goal is set by an orchestration layer and interpreted by scheduler layer.
   * In the end it is used by low-level scheduler to make scheduling decisions e.g. should the task associated with this instance be launched or killed?
   */
-sealed trait Goal extends Product with Serializable
+sealed trait Goal extends Product with Serializable {
+
+  /**
+    * A doomed instance's task will be killed (eventually)
+    */
+  def isDoomed(): Boolean = this match {
+    case _: Goal.Doomed => true
+    case _ => false
+  }
+}
 
 object Goal {
 
+  sealed trait Doomed extends Goal
```

Comment:
Let the naming battle begin! 😄

Reply:
I used `doomed` as well so that is good. See my comment below.